prowler/providers/aws/services/ec2/ec2_instance_not_directly_publicly_accessible_via_elb/ec2_instance_not_directly_publicly_accessible_via_elb.py

```diff
@@ -0,0 +1,31 @@
+from prowler.lib.check.models import Check, Check_Report_AWS
+from prowler.providers.aws.services.ec2.ec2_client import ec2_client
+from prowler.providers.aws.services.elb.elb_client import elb_client
+
+
+class ec2_instance_not_directly_publicly_accessible_via_elb(Check):
+    def execute(self):
+        findings = []
+
+        public_instances = {}
+        for lb in elb_client.loadbalancers:
+            if lb.public:
+                for instance in lb.instances_ids:
+                    public_instances[instance] = lb
+
+        for instance in ec2_client.instances:
+            if instance.state != "terminated":
+                report = Check_Report_AWS(self.metadata())
+                report.region = instance.region
+                report.resource_id = instance.id
+                report.resource_arn = instance.arn
+                report.resource_tags = instance.tags
+                report.status = "PASS"
+                report.status_extended = f"EC2 Instance {instance.id} is not publicly accesible through an Internet facing Classic Load Balancer."
+
+                if instance.id in public_instances and instance.security_groups <= 1:
```

Comment:
The security group must not be used in this way, here you are only checking that it exits but not if it has any inbound rule

Reply:
done